tests/test_model_specs.py

```diff
@@ -6,47 +6,111 @@
 
 import pint
 from natcap.invest.models import model_id_to_pyname
+from natcap.invest import spec
 from osgeo import gdal
 
 PLUGIN_URL = 'git+https://github.com/emlys/demo-invest-plugin.git'
 PLUGIN_NAME = 'foo-model'
 
 
 gdal.UseExceptions()
-valid_nested_types = {
+valid_nested_input_types = {
     None: {  # if no parent type (arg is top-level), then all types are valid
-        'boolean',
-        'integer',
-        'csv',
-        'directory',
-        'file',
-        'freestyle_string',
-        'number',
-        'option_string',
-        'percent',
-        'raster',
-        'ratio',
-        'vector',
+        spec.IntegerInput,
+        spec.NumberInput,
+        spec.RatioInput,
+        spec.PercentInput,
+        spec.StringInput,
+        spec.OptionStringInput,
+        spec.BooleanInput,
+        spec.SingleBandRasterInput,
+        spec.VectorInput,
+        spec.RasterOrVectorInput,
+        spec.CSVInput,
+        spec.DirectoryInput,
+        spec.FileInput
```

Comment:
Annoying request, can we alphabetize these?

Reply:
And within the grouped lists below?

Reply:
Done in 57f91dfd8